### DISCUSSION TITLE
Is there any example code about bipartite graph?

### DISCUSSION THREAD

**Discussion by Anya0101** [pyg-team/pytorch_geometric#4109] [Q&A] [answered]
Hi,
I am now having a project with a bipartite graph to be trained. Though there are documents such as [how to build the bipartite data](https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html?#bipartite-graphs), and lots of discussions in this repository, I am still wondering if there is an example code about bipartite graph from data preprocessing to training and then testing? Thanks!

**Comment by rusty1s** [comment, marked as answer]
See https://pytorch-geometric.readthedocs.io/en/latest/notes/load_csv.html and https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hetero_link_pred.py.
